### DISCUSSION TITLE
Per-Server API Key

### DISCUSSION THREAD

**Discussion by heisreald** [Kav-K/GPTDiscord#116] [Ideas]
Hey Kav! I'm loving following the development of this project. I started using GPT3Discord in December in some private servers. Recently, there seems to be more demand and I've updated to 8.4.1, starting DeepL integration today. I like the per-user API key feature, but I was wondering if this can be improved to add per-server keys/per-guildID? I'm developing the bot to be used in multiple servers and the per-server key would help to keep things organized. If I can get going with this, I would love to sponsor the project financially.

Thanks for your work!

**Comment by Kav-K** [comment]
Hi, we've only implemented per-user API keys, do you still have a use for per-server API keys?

**Reply by slacksystem** [reply]
I'm not from the initial post but I feel like per-server keys could be very helpful.

**Reply by Kav-K** [reply]
Curious, what use case would you have for it? I just dont understand why people would want different API keys for different servers, as it seems like a lot of overhead work, no?
